backend/app/routes/admin/resetPassword.js

```diff
@@ -0,0 +1,52 @@
+const argon2 = require('argon2');
+const Admin = require('../../models/Admin');
+const { ErrorHandler } = require('../../../helpers/error');
+const constants = require('../../../constants');
+const { verifyToken } = require('../../../helpers/middlewares/auth');
+
+module.exports = async (req, res, next) => {
+  // Extracting the new password from the request
+  const { newPassword } = req.body;
+  try {
+    // Extracting the TOKEN from URl /forgotpassword/:token
+    const { token } = req.params;
+
+    // Verifying the token and extracting the email from the payload
+    const verifiedToken = await verifyToken(token);
+    const { email } = verifiedToken;
+
+    // Finding the admin that email
+    const userRecord = await Admin.findOne({ email });
```

Comment:
Please use a single DB call for the flow. DB calls are usually quite expensive and becomes a potential contributor to overall latency